### ISSUE TITLE
android warnings

### ISSUE THREAD

**Issue by AlexB123** [syzygy1/Cfish#88]
Hello mr.Ronald!
Can you take a look on the Cfish code? I'm asking you because, during arm7 and arm8 engines for android i'm getting some warnings (engines compiles, but they crashes often times in analysis mode).
Best regards.
Alex.
---------------------------------------------------
Generating 64 bit engine for ARM-Android.
movegen.c:346:1: warning: unknown attribute 'optimize' ignored
      [-Wunknown-attributes]
SMALL
^
./types.h:62:30: note: expanded from macro 'SMALL'
#define SMALL __attribute__((optimize("Os")))
                             ^
1 warning generated.
movepick.c:77:1: warning: unknown attribute 'optimize' ignored
      [-Wunknown-attributes]
SMALL
^
./types.h:62:30: note: expanded from macro 'SMALL'
#define SMALL __attribute__((optimize("Os")))
                             ^
1 warning generated.
position.c:1335:16: warning: unknown attribute 'optimize' ignored
      [-Wunknown-attributes]
__attribute__((optimize("Os")))
               ^
1 warning generated.
Generating 32 bit engine for ARM-Android.
movegen.c:346:1: warning: unknown attribute 'optimize' ignored
      [-Wunknown-attributes]
SMALL
^
./types.h:62:30: note: expanded from macro 'SMALL'
#define SMALL __attribute__((optimize("Os")))
                             ^
1 warning generated.
movepick.c:77:1: warning: unknown attribute 'optimize' ignored
      [-Wunknown-attributes]
SMALL
^
./types.h:62:30: note: expanded from macro 'SMALL'
#define SMALL __attribute__((optimize("Os")))
                             ^
1 warning generated.
position.c:1335:16: warning: unknown attribute 'optimize' ignored
      [-Wunknown-attributes]
__attribute__((optimize("Os")))
               ^
1 warning generated.
Press any key to continue . . .

**Comment by AlexB123** [comment]
Sorry, here is the correct warnings, some arrows "^" are in wrong place in my previous post.
------------------------------------------------------------------------------

![screenshot](https://user-images.githubusercontent.com/19660133/53057781-6800fc80-34b9-11e9-92e5-1b547ac4c541.jpg)

**Comment by syzygy1** [comment]
These warnings cannot lead to crashes.

**Comment by AlexB123** [comment]
Thank you! Probably you're right, but still, after 10 engine runs (in analysis mode), there was 4 crashes. I guess, it is some kind of Clang bug or something.

**Comment by syzygy1** [comment]
There may of course be a bug, but the warnings don't cause crashes.

Did the crashes start with the last update or have they happened for longer? (The warnings must have been there since a very long time, btw. But again, they are certainly unrelated.)

**Comment by AlexB123** [comment]
You're right, those warnings was since a very long time, but the engines crashes started around after "Updated to "Introduce Multi-Cut". (again i'm not 100% sure). Regarding the warnings - a friend of mine says that, they appearing because clang don't support the specific gcc code, here is the solution (on picture). The question is, can these changes affect the behavior of the engine?
![1](https://user-images.githubusercontent.com/19660133/53293110-6c3d5a80-37d6-11e9-976f-a48389cd2788.jpg)
==============================================================
![2](https://user-images.githubusercontent.com/19660133/53293114-795a4980-37d6-11e9-8448-43b18281453d.jpg)

**Comment by syzygy1** [comment]
Do the crashes still happen?

The reason for SMALL is to avoid inlining. If you can't get it to compile, you can #define it to nothing or even just remove all occurrences of SMALL. However, __attribute__((minsize)) would seem to do exactly what is intended. You could test with and without __attribtue__((minsize)) to see if there is a difference in speed.

**Comment by syzygy1** [comment]
I have now fixed the warnings.

Do the crashes still happen?
